client/src/components/Overview/AddToCart.jsx

```diff
@@ -69,74 +66,95 @@ const AddToCart = () => {
   };
 
   // SIZE SELECTOR ========================================================
+  // when a size is selected, show quantity dropdown and close size dropdown
   const handleSizeInput = (inputSize) => {
+    setSize(inputSize);
     if (inputSize === 'Select Size') {
       showQty(false);
       setSku(availableSkus[0]);
     } else {
-      setSize(inputSize);
       const matchingSku = availableSkus.find((sku) => inputSize === sku[1].size);
       setSku(matchingSku);
       showQty(true);
-      openSizes(false);
+      showSizes(false);
+      showError(false);
     }
   };
 
+  // find the sizes for skus in stock
   const availableSizes = availableSkus.map(
     (sku) => (
-      <span
+      <li
         key={sku[0]}
-        className="sizes"
+        className="size"
         role="menuitem"
         tabIndex="-1"
         onClick={() => handleSizeInput(sku[1].size)}
         onKeyPress={() => handleSizeInput(sku[1].size)}
       >
         {sku[1].size}
-      </span>
+      </li>
     ),
   );
 
-  const renderSizeSelecter = () => {
+  const renderSizeSelector = () => {
```

Comment:
the drop down menu for the size selector opens on hover instead of on click, feature or bug?